components/Common/OramaSearchBar/index.tsx

```diff
@@ -0,0 +1,91 @@
+import { create, load, search } from '@orama/orama';
+import { useTheme } from 'next-themes';
+import { useCallback, useMemo, useState } from 'react';
+import { FormattedMessage } from 'react-intl';
+import { OramaLogoDark, OramaLogoLight } from './OramaLogo';
+import styles from './index.module.scss';
+import SearchBar from '../SearchBar';
+
+import type { Document, Orama, RawData, Schema } from '@orama/orama';
+import type { FC } from 'react';
+import type { SearchFunction, SearchResult } from '../../../types';
+
+type DocumentMapper = (doc: Document) => SearchResult;
+
+type FooterProps = {
+  theme?: string;
+};
+
+type OramaSearchBarProps = {
+  schema: Schema;
+  index: RawData;
+  documentMapper?: DocumentMapper;
+};
+
+function identityDocumentMapper(doc: Document): SearchResult {
+  return doc as unknown as SearchResult;
+}
+
+async function performSearch(
+  database: Orama,
+  mapper: DocumentMapper,
+  term: string
+): Promise<SearchResult[]> {
+  const results = await search(database, { term });
+
+  return results.hits.map(h => mapper(h.document));
+}
+
+const Footer: FC<FooterProps> = ({ theme }: FooterProps) => {
+  return (
+    <div className={styles.oramaSearchBarFooter}>
+      <span className={styles.oramaSearchBarFooterLabel}>
+        <FormattedMessage id="components.oramaSearchBar.search.poweredBy" />
+      </span>
+      <a
+        href={'http://oramajs.io'}
+        target="_blank"
+        rel="noopener noreferrer"
+        className={styles.oramaSearchBarFooterLink}
+      >
+        {theme === 'dark' ? (
```

Comment:
This can be memoized I guess. Also probably you want to use `useTheme` hook for the theme :)

Reply:
I was already using `useTheme` but in the parent component for another stupid reason I'm to embarassed to mention now.
Anyway this is fixed now. :)